src/components/scheduling/goals/SchedulingGoal.svelte

```diff
@@ -42,6 +53,32 @@
     simulateGoal = goalPlanSpec.simulate_after; // Copied to local var to reflect changed values immediately in the UI
   }
 
+  $: {
+    let revision: number | null = null;
+    if (goalPlanSpec.goal_revision == null) {
+      revision = Math.max(...(goalPlanSpec.goal_metadata?.versions?.map(x => x.revision) ?? [0]));
+    } else {
+      revision = goalPlanSpec.goal_revision;
+    }
+    version = goalPlanSpec.goal_metadata?.versions.filter(x => x.revision === revision)[0];
+    let schema = version?.parameter_schema;
+    let result: any[] = [];
+    if (schema && schema.type === 'struct') {
+      Object.entries(schema.items).forEach(([name, subschema], i) => {
```

Comment:
Rather than pushing into a `result` array typed as `any`, can this be structured like
```
if (schema && schema.type === 'struct') {
   formParameters = Object.entries(schema.items).map([name, subschema], i) => { 
     ...
    return { ... };
   });
} else {
  formParameters = [];
}
```